api/v2/datadog/api_logs.go

```diff
@@ -216,21 +241,76 @@ func (r *ListLogsOptionalParameters) WithBody(body LogsListRequest) *ListLogsOpt
  * [2]: https://docs.datadoghq.com/logs/archives
  */
 func (a *LogsApiService) ListLogs(ctx _context.Context, o ...ListLogsOptionalParameters) (LogsListResponse, *_nethttp.Response, error) {
-	req := apiListLogsRequest{
-		ApiService: a,
-		ctx:        ctx,
-	}
-
-	if len(o) > 1 {
+	req, err := a.buildListLogsRequest(ctx, o...)
+	if err != nil {
 		var localVarReturnValue LogsListResponse
-		return localVarReturnValue, nil, reportError("only one argument of type ListLogsOptionalParameters is allowed")
+		return localVarReturnValue, nil, err
 	}
 
-	if o != nil {
-		req.body = o[0].Body
+	return req.ApiService.listLogsExecute(req)
+}
+
+/*
+ * ListLogsWithPagination provides a paginated version of ListLogs returning a channel with all items.
+ */
+func (a *LogsApiService) ListLogsWithPagination(ctx _context.Context, o ...ListLogsOptionalParameters) (items chan Log, cancel func(), err error) {
+	ctx, cancel = _context.WithCancel(ctx)
+	pageSize_ := int32(10)
+	if len(o) > 0 && o[0].Body.Page.Limit != nil {
```

Comment:
There is an issue here, because one of Body or Page can be nil, so this would segfault. We need to create intermediate data structures if they are not present.